server/test/views/components/TextFormatterTest.java

```diff
@@ -170,14 +170,14 @@ public void listRendersCorrectly() {
   }
 
   @Test
-  public void rendersRecursively() {
+  public void rendersRecursivelyWithMarkdown() {
     String text =
         "Cheesecake Recipe\n"
```

Comment:
Can you include a test case for markdown formatted urls. ```[link text](the_hyperlink)```. This has been a huge request from folks in Seattle.

Reply:
Also please add a test case that has `<script>alert('bad-time');</script>` in the markdown and verifies that there is no script tag in the generated html.

Reply:
The url one is a tricky one. The current logic in `createLinksAndEscapeText` detects urls automatically, and I implemented the markdown parsing so it only applies to strings surrounding the urls, not the urls themselves. Can you tell me more about the need for markdown formatted urls though? Have folks in Seattle wanted urls to be formatted differently that they are now with the automatic detection?

Reply:
Yeah they have. Being able to have custom link text, something friendlier than a long "scary" url, gets brought up often. I always point at #888 so duplicate issues don't get filed.

Reply:
That's good to know and complicates things a little more. Since it all runs through `formatText` we would need a way to distinguish if folks want the "scary" urls, or the markdown ones, which would involve a UI change. Unless we want to apply a rule based on where the url is rendering (program description vs. confirmation screen vs. static question) or we want to do away with scary urls all together which I could talk with Sijia about. What do you think?

Reply:
Update that Gwen and I talked this over and decided that the optimal solution is to do away with the custom auto-detecting functionality and just use Markdown URL syntax. This would give admins the ability to format text and URLs which is something they've been asking for! I want to make sure we maintain the ability to auto-detect URLs so I'll look into that before deprecating this custom logic.